libvirt.go

```diff
@@ -80,6 +83,34 @@ func NewVirConnection(uri string) (VirConnection, error) {
 	return obj, nil
 }
 
+func NewVirConnectionWithAuth(uri string, username string, password string) (VirConnection, error) {
```

Comment:
SASL has more than just these two mechs.
Maybe I should be more specific in the method name that this is about username and password authentication? 
Of course we could work here with callbacks too ... Thoughts?